filters/context.go

```diff
@@ -114,6 +122,32 @@ func (ctx *ctext) IsMITMing() bool {
 	return mitming != nil && mitming.(bool)
 }
 
+func (ctx *ctext) Value(key interface{}) interface{} {
+	values := ctx.getLongLivedValues()
+	// try to get key/value from our map
+	value, found := values[key]
+	if !found {
+		// no value found in map, look in wrapped Context
+		value = ctx.Context.Value(key)
+		if value == nil {
+			value = nilValue
```

Comment:
Why can't `value` just remain `nil`?

Reply:
Then instead of
```
	if value == nilValue {
		return nil
	}
	return value
```
at the end of this function, you could just have `return value`.  Right?  What am I missing?

Reply:
Because then we would always end up trying to look up the value from the Context again.

Reply:
In that case, `found` should evaluate to `true` and we would just return nil.

https://play.golang.org/p/WzQQUXl7vaD